### PR TITLE
Allow configuring SecurityContextRepository for BasicAuthenticationFilter

### PR DESCRIPTION
**Pull request by evgeniycheban**
Allow configuring SecurityContextRepository for BasicAuthenticationFilter

Closes gh-12031

### CODE REVIEW

File: config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java [hunk 1]
```diff
@@ -192,6 +212,7 @@ public void configure(B http) {
 		AuthenticationManager authenticationManager = http.getSharedObject(AuthenticationManager.class);
 		BasicAuthenticationFilter basicAuthenticationFilter = new BasicAuthenticationFilter(authenticationManager,
 				this.authenticationEntryPoint);
+		basicAuthenticationFilter.setSecurityContextRepository(this.securityContextRepository);
```
Comment (sjohnr) [inline config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java:215]: I'm wondering if it makes more sense to provide no default for `securityContextRepository` in this configure (therefore it defaults to `null`), and only set on the filter if not `null`?

**Review by sjohnr — COMMENTED** [review]
Thanks @evgeniycheban! I've added on review comment below.

File: config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java (hunk 1 above)
Comment (evgeniycheban) [reply config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java:215]: Assigning a default value and performing a `null` check in the setter seems like a common strategy that is used elsewhere, so I would stay here with this approach and, perhaps, later align other methods in this class.

Comment (sjohnr) [reply config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java:215]: Hi @evgeniycheban! Sorry for the delay in responding.

> Assigning a default value and performing a `null` check in the setter seems like a common strategy that is used elsewhere

I was thinking specifically of configurers where it seems the pattern is often to let the configured `Filter` contain the default and only override the filter (via the setter) if not null.

Can you share an example in other configurers where you see this strategy used? Or are you referring mostly to other classes (besides configurers) in Spring Security?

Comment (evgeniycheban) [reply config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java:215]: @sjohnr yes, I was referring to other classes where this pattern is commonly used.

Comment (evgeniycheban) [reply config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java]: @sjohnr I've updated the PR to remove the default value and add a `null` check.

Comment (sjohnr) [reply config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java]: Awesome! I’ll work on getting this merged shortly.

**Comment by sjohnr** [comment]
Thanks @evgeniycheban! Merged via 59829321a8045c26f19c725f3ae081ac297e4079

**Pull request closed without merging**